apis/tempo/v1alpha1/tempostack_webhook.go

```diff
@@ -316,6 +319,20 @@ func (v *validator) validateTenantConfigs(tempo TempoStack) field.ErrorList {
 	return nil
 }
 
+func (v *validator) validateStackName(tempo TempoStack) field.ErrorList {
+	// We need to check this because the name is used as a label value for app.kubernetes.io/instance
+	// Only validate the length, because the DNS rules are validated by kubernetes
```

Comment:
```suggestion
	// Only validate the length, because the DNS rules are enforced by the functions in the `naming` package.
```
afaics with this PR it's impossible to generate invalid names, as the length is checked here in the validator and the `DNSName()` replaces invalid characters.